camel-core/src/main/java/org/apache/camel/builder/TransformerBuilder.java

```diff
@@ -0,0 +1,184 @@
+        this.scheme = scheme;
+        return this;
+    }
+
+    /**
+     * Set the 'from' data type .
+     *
+     * @param from 'from' data type
+     */
+    public TransformerBuilder from(String from) {
+        this.from = from;
+        return this;
+    }
+
+    /**
+     * Set the 'from' data type using Java class.
+     *
+     * @param clazz 'from' Java class
+     */
+    public TransformerBuilder from(Class<?> from) {
+        this.from = "java:" + from.getName();
+        return this;
+    }
+
+    /**
+     * Set the 'to' data type .
+     *
+     * @param to 'to' data type
+     */
+    public TransformerBuilder to(String to) {
+        this.to = to;
+        return this;
+    }
+
+    /**
+     * Set the 'to' data type using Java class.
+     *
+     * @param clazz 'to' Java class
+     */
+    public TransformerBuilder to(Class<?> to) {
+        this.to = "java:" + to.getName();
+        return this;
+    }
+
+    /**
+     * Set the URI to be used for the endpoint {@code Transformer}.
+     * @see {@code EndpointTransformerDefinition}, {@code ProcessorTransformer}
+     * 
+     * @param uri endpoint URI
+     */
+    public TransformerBuilder withUri(String uri) {
+        resetType();
+        this.uri = uri;
+        return this;
+    }
+
+    /**
+     * Set the {@code DataFormatDefinition} to be used for the {@code DataFormat} {@code Transformer}.
+     * @see {@code DataFormatTransformerDefinition}, {@code DataFormatTransformer}
+     * 
+     * @param dfd {@code DataFormatDefinition}
+     */
+    public TransformerBuilder withDataFormat(DataFormatDefinition dfd) {
+        resetType();
+        this.dataFormat = dfd;
+        return this;
+    }
+
+    /**
+     * Set the Java {@code Class} represents a custom {@code Transformer} implementation class
+     * to be used for custom Transformer.
+     * @see {@code CustomTransformerDefinition}
+     * 
+     * @param clazz {@code Class} object represents custom transformer implementation
+     */
+    public TransformerBuilder withJava(Class<? extends Transformer> clazz) {
+        resetType();
+        this.clazz = clazz;
+        return this;
+    }
+
+    /**
+     * Set the Java Bean name to be used for custom {@code Transformer}.
+     * @see {@code CustomTransformerDefinition}
+     * 
+     * @param ref bean name for the custom {@code Transformer}
+     */
+    public TransformerBuilder withBean(String ref) {
+        resetType();
+        this.beanRef = ref;
+        return this;
+    }
+
+    private void resetType() {
+        this.uri = null;
+        this.dataFormat = null;
+        this.clazz = null;
+        this.beanRef = null;
+    }
+
+    /**
+     * Configure a Transformer according to the configurations built on this builder
+     * and register it into given {@code CamelContext}.
+     * 
+     * @param camelContext {@code CamelContext}
+     */
+    public void configure(CamelContext camelContext) {
+        TransformerDefinition transformer;
```

Comment:
What do you think to extract this big block of if then else in a private method "createTransformerDefinition" ?